### PR TITLE
Add redirects for all existing plugins to redirect to plugin site

### PR DESCRIPTION
**Pull request by halkeye**
In theory this will allow wiki exporter to come through, but all the rest will redirect to plugins site.

Okay, to prevent jenkins-wiki-exporter from being unable to export existing pages, this PR feels a bit ugly, but I got a nice set of rules that redirect all plugin docs pages to plugins site.
Any existing migrated plugins will not be in the list, cause https://updates.jenkins.io/current/plugin-documentation-urls.json already has them updated.

Generated with:
```
/* eslint-disable no-console */
const axios = require('axios');
const escapeStringRegexp = require('escape-string-regexp');
const {parse:parseUrl} = require('url');

const requestGET = (url) => {
    return axios
        .get(url)
        .then((results) => {
            if (results.status !== 200) {
                throw results.data;
            }
            return results.data;
        });
};
const pluginWikiUrlRe = /^https?:\/\/wiki.jenkins(?:-ci.org|.io)\/display\/(?:jenkins|hudson)\/([^/]*)\/?$/i;

requestGET('https://updates.jenkins.io/current/plugin-documentation-urls.json').then(urls => {
  for (const [key, {url}] of Object.entries(urls)) {
    if (!pluginWikiUrlRe.test(url)) { continue; }
    const [other, page] = url.match(pluginWikiUrlRe);
    console.log(`RewriteCond %{HTTP_USER_AGENT} !^jenkins-wiki-exporter/(.*)$`)
    console.log(`RewriteRule "^${escapeStringRegexp(parseUrl(other).pathname).replace(/\+/g, '\+')}$" "https://plugins.jenkins.io/${key}" [NC,L,QSA,R=301]`);
  }
});
```

### CODE REVIEW

**Review by timja — APPROVED** [review]

File: dist/profile/templates/confluence/vhost.conf [hunk 1]
```diff
@@ -6,6 +6,2813 @@ RewriteEngine on
...
+RewriteCond %{HTTP_USER_AGENT} !^jenkins-wiki-exporter/(.*)$
+RewriteRule "^/display/JiraTestResultReporter\-plugin$" "https://plugins.jenkins.io/JiraTestResultReporter" [NC,L,QSA,R=301]
+RewriteCond %{HTTP_USER_AGENT} !^jenkins-wiki-exporter/(.*)$
+RewriteRule "^/display/Matrix\+configuration\+sorter\+plugin$" "https://plugins.jenkins.io/Matrix-sorter-plugin" [NC,L,QSA,R=301]
+RewriteCond %{HTTP_USER_AGENT} !^jenkins-wiki-exporter/(.*)$
+RewriteRule "^/display/Office\+365\+Connector\+Plugin$" "https://plugins.jenkins.io/Office-365-Connector" [NC,L,QSA,R=301]
+RewriteCond %{HTTP_USER_AGENT} !^jenkins-wiki-exporter/(.*)$
+RewriteRule "^/display/Parameterized\+Remote\+Trigger\+Plugin$" "https://plugins.jenkins.io/Parameterized-Remote-Trigger" [NC,L,QSA,R=301]
+RewriteCond %{HTTP_USER_AGENT} !^jenkins-wiki-exporter/(.*)$
+RewriteRule "^/display/Priority\+Sorter\+Plugin$" "https://plugins.jenkins.io/PrioritySorter" [NC,L,QSA,R=301]
```
Comment (olblak) [inline dist/profile/templates/confluence/vhost.conf]: Are you not missing /JENKINS/ between /display and plugin name?
An exampled `https://wiki.jenkins.io/display/JENKINS/Priority+Sorter+Plugin`

**Review by olblak — CHANGES_REQUESTED** [review]
RewriteRule url need to be fixed, as plugin urls on wiki are not correct

**Comment by halkeye** [comment]
thanks @olblak that was super dumb of me. I updated the script and generating the right urls now

**Review by MarkEWaite — APPROVED** [review]

**Review by rtyler — APPROVED** [review]
Looks good to me, puppet will need to be re-enabled on the host to deploy this

**Comment by olblak** [comment]
> Looks good to me, puppet will need to be re-enabled on the host to deploy this
I don't recall or see a reason why puppet was disabled on lettuce

**Comment by timja** [comment]
> > Looks good to me, puppet will need to be re-enabled on the host to deploy this
> > I don't recall or see a reason why puppet was disabled on lettuce

I think @rtyler re-enabled it, see IRC

**Review by olblak — APPROVED** [review]

**Comment by timja** [comment]
I think the auto prod PR function is broken atm too

**Pull request merged**